homeassistant/components/mqtt/__init__.py

```diff
@@ -107,12 +106,11 @@ def valid_publish_topic(value):
     }),
 }, extra=vol.ALLOW_EXTRA)
 
-MQTT_BASE_PLATFORM_SCHEMA = vol.Schema({
-    vol.Required(CONF_PLATFORM): DOMAIN,
-    vol.Optional(CONF_SCAN_INTERVAL):
-        vol.All(vol.Coerce(int), vol.Range(min=1)),
+SCHEMA_BASE = {
     vol.Optional(CONF_QOS, default=DEFAULT_QOS): _VALID_QOS_SCHEMA,
-})
+}
+
+MQTT_BASE_PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend(SCHEMA_BASE)
```

Comment:
For a future PR, but we should really kill this. It was a bad design decision from my side.